mtags/src/main/scala-2/scala/meta/internal/pc/ExtractMethodProvider.scala

```diff
@@ -0,0 +1,138 @@
+package scala.meta.internal.pc
+
+import scala.reflect.internal.util.Position
+
+import scala.meta.internal.mtags.MtagsEnrichments._
+import scala.meta.internal.pc.ExtractMethodUtils
+import scala.meta.pc.OffsetParams
+import scala.meta.pc.RangeParams
+
+import org.eclipse.{lsp4j => l}
+
+final class ExtractMethodProvider(
+    val compiler: MetalsGlobal,
+    range: RangeParams,
+    extractionPos: OffsetParams
+) extends ExtractMethodUtils {
+  import compiler._
+  def extractMethod: List[l.TextEdit] = {
+    val text = range.text()
+    val params = range.toOffset
+    val unit = addCompilationUnit(
+      code = text,
+      filename = params.uri.toString(),
+      cursor = None
+    )
+    val pos = unit.position(params.offset)
+    val context = doLocateImportContext(pos)
+    val re: scala.collection.Map[Symbol, Name] = renamedSymbols(context)
+    typedTreeAt(pos)
+    val history = new ShortenedNames(
+      lookupSymbol = name =>
+        context.lookupSymbol(name, sym => !sym.isStale) :: Nil,
+      config = renameConfig,
+      renames = re
+    )
+    def prettyType(tpe: Type) =
+      metalsToLongString(tpe.widen.finalResultType, history)
+
+    val rangeSourcePos = Position.range(
```

Comment:
Do you need to convert it here? We could just change the `encloses` method to work with RangeParams